CONTRIBUTING.md

```diff
@@ -0,0 +1,223 @@
+- All tests pass
+
+## Introduce a New Loader
+
+If you want to introduce a new loader, please follow the steps below:
+
+- Open an issue to discuss the new loader
+  - Describe the use case for the new loader
+	- What problem does it solve?
+	- What are the benefits of the new loader?
+	- Which dependencies will be required?
+- Once the issue is approved, make changes to the codebase
+- Open a pull request
+- Once the pull request is approved, the new loader will be merged
+- Let's realease it.
+
+### Create a New Loader
+
+1. Create a new directory in the `packages` directory with the name of the loader
+2. Create the package.json file
+```json
+{
+	"version": "1.0.0",
+	"name": "@nodejs-loaders/your-loader",
+	"type": "module",
+	"author": "Your Name",
+	"maintainers": [
+		"Jacob Smith",
+		"Augustin Mauroy"
+	],
+  "license": "ISC",
+	"main": "./your-loader.mjs",
+  "repository": {
+    "url": "https://github.com/JakobJingleheimer/nodejs-loaders"
+  }
+}
+```
+
+> [!NOTE]
+> The `repository.url` field must be present. It's use for generating the provenance signature.
+
+3. Create the loader
+
+Your loader have to export [customisaiton hooks](https://nodejs.org/api/module.html#customization-hooks) to be able to customize the loader behavior.
+
+```js
+/**
+ * @type {import('node:module').LoadHook}
+ */
+function resolveMyLoader(specifier, ctx, next) {
+	/* Do some awesome stuff */
+}
+export { resolveMyLoader as resolve };
+
+/**
+ * @type {import('node:module').LoadHook}
+ */
+function loadMyLoader(url, ctx, defaultLoad) {
+	/* Do some awesome stuff */
+}
+export { loadMyLoader as load };
+```
+
+4. Write unit tests
+
+Yours customisation hooks must be tested. There are two part of testing, unit tests and integration tests.
+
+First write unit tests for your hooks. We use [`node:test`](https://nodejs.org/api/test.html) to run the tests. All unit test should have this naming convention `your-loader.spec.mjs`. And the test file should be in the same directory as the loader. Also the test file should have the following structure:
+
+```js
+import assert from 'node:assert/strict';
+import { describe it } from 'node:test';
+
+import { resolve, load } from './your-loader.mjs';
+
+describe('Your Loader', () => {
+	describe('resolve', () => {
+		it('should resolve the specifier', () => {
+			/* Test your resolve hook */
+		});
+	});
+
+	describe('load', () => {
+		it('should load the module', () => {
+			/* Test your load hook */
+		});
+	});
+});
+```
+
+Second write integration tests for your loader. We use [`node:module`](https://nodejs.org/api/module.html) to load the loader.
+
+```js
+import assert from 'node:assert/strict';
+import { spawnSync } from 'node:child_process';
+import { execPath } from 'node:process';
+import { describe, it } from 'node:test';
+import { fileURLToPath } from 'node:url';
+
+describe('Your Loader', () => {
+	it('should work with --loader', () => {
+		const { status, stdout, stderr } = spawnSync(execPath, [
```

Comment:
Should use `spawnPromisified` for anything not a benchmark :)

Reply:
honestly idk maybe